contracts/truefi2/LineOfCreditAgency.sol

```diff
@@ -249,9 +249,15 @@ contract LineOfCreditAgency is UpgradeableClaimable, ILineOfCreditAgency {
      * @dev Internal function to update `borrower` credit score for `pool` using credit oracle
      * @return Tuple containing (oldScore, newScore)
      */
-    function _updateCreditScore(ITrueFiPool2 pool, address borrower) internal returns (uint8, uint8) {
+    function _updateCreditScore(
+        ITrueFiPool2 pool,
+        address borrower,
+        uint256 borrowedAmount
+    ) internal returns (uint8, uint8) {
         uint8 oldScore = creditScore[pool][borrower];
         uint8 newScore = creditOracle.score(borrower);
+        uint256 stakedAmount = collateralVault.stakedAmount(borrower);
+        newScore = creditModel.effectiveScore(newScore, pool, stakedAmount, borrowedAmount);
         creditScore[pool][borrower] = newScore;
```

Comment:
nitpick: We should consistently calling this `effectiveScore` -- Roshan wants to make it clear that borrower staking does not change actual credit scores. Can be done in a follow-up PR.

Reply:
Do You mean we should not store it in LoCA? We still do have our Credit Oracle, which would remain as the source of truth.

Reply:
My bad for the confusion. I just meant that the local variable in this function should be named `effectiveScore` instead of `newScore` after we calculate effective score. It's the tiniest of nitpicks, mostly irrelevant and unimportant to address.